### PR TITLE
Yc je issue eight

### PR DESCRIPTION
**Pull request by AndrewEspejo**
This PR add principally a new functionality, the user can see through a checkbox if he/she has purchased an item in the last 24 hours

## Description

At first we had some problems with the methodology for handling the dates in React. But, we were able to solve it by bringing the dates from Firebase and applying a method to be able to compare it with another date and achieve the checkbox deactivation event after 24 hours.

## Acceptance Criteria

- [ ] User is able to tap a checkbox or similar UI element to mark an item in the list as purchased

- [ ] Item should be shown as checked for 24 hours after the purchase is made (i.e. we assume the user does not need to buy the item again for at least 1 day)

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|  ✓ | :sparkles: New feature     |

### Before

Selection_012

### After

Selection_013

## Testing Steps / QA Criteria
If you wanna testing this new functionality, first clone the repo and install node modules, after that you need to create a new token , later add an item to your list and that is all. You can see the box in an active state as long as 24 hours pass

### CODE REVIEW

File: src/components/Product/ListProduct.js [hunk 1]
```diff
@@ -1,6 +1,11 @@
 import React from 'react';
 import { FirestoreCollection } from 'react-firestore';
+import { differenceInDays } from 'date-fns';
+
 const ListProduct = () => {
+  const lastDate = new Date();
+  const currentDateSeconds = lastDate.getTime() / 1000;
```
Comment (alejandronanez) [inline src/components/Product/ListProduct.js]: 👍 Nice name for this variable. Good work,

File: src/components/Product/Product.js [hunk 2]
```diff
@@ -39,7 +39,7 @@ function Product() {
     colecction
       .add({
         name: nameProduct,
-        lastDate: data.lastDate,
+        lastDate: new Date(),
```
Comment (alejandronanez) [inline src/components/Product/Product.js]: This is not completely correct.

We should default to `new Date()` only if the user didn't add anything in the form. I think about something like:

```
lastDate: data.lastDate || new Date()
```

File: src/components/Product/ListProduct.js [hunk 3]
```diff
@@ -17,13 +22,23 @@ const ListProduct = () => {
...
-                  <td>{value.name}</td>
-                  <td>{value.date} </td>
-                  <td>{value.lastDate}</td>
-                </tr>
-              ))}
+              {data.map((value, key) => {
+                const differenceDays = differenceInDays(
+                  value.lastDate.seconds,
+                  currentDateSeconds,
+                );
```
Comment (alejandronanez) [inline src/components/Product/ListProduct.js]: This is not working as expected, my guess is that `value.lastDate.seconds` is returning `undefined`.

What if you store the date when the product was last added, in seconds instead?

**Review by alejandronanez — CHANGES_REQUESTED** [review]
@AndrewEspejo @Yesika-Castillo 👋 Left a couple of thoughts for y'all.

File: src/components/Product/ListProduct.js (hunk 1 above)
Comment (Yesika-Castillo) [reply src/components/Product/ListProduct.js]: thanks

File: src/components/Product/ListProduct.js (hunk 3 above)
Comment (Yesika-Castillo) [reply src/components/Product/ListProduct.js]: the issue was fixed

File: src/components/Product/Product.js (hunk 2 above)
Comment (Yesika-Castillo) [reply src/components/Product/Product.js]: done!!!

File: src/components/Product/ListProduct.js [hunk 4]
```diff
@@ -17,13 +20,24 @@ const ListProduct = () => {
...
-                  <td>{value.name}</td>
-                  <td>{value.date} </td>
-                  <td>{value.lastDate}</td>
-                </tr>
-              ))}
+              {data.map((value, key) => {
+                const differenceDays =
+                  (currentDateSeconds -
+                    (!!value.lastDate && !!value.lastDate.seconds
+                      ? value.lastDate.seconds
```
Comment (DNeosphere) [inline src/components/Product/ListProduct.js]: :clap:

**Review by alejandronanez — APPROVED** [review]
Nice work y'all ✅

**Comment by alejandronanez** [comment]
@AndrewEspejo @Yesika-Castillo make sure to resolve the conflict you have in `ListProduct` before merging.

Nice work!

**Pull request merged**